mesos/mesos.go

```diff
@@ -239,12 +258,14 @@ func (m *Mesos) CollectMetrics(mts []plugin.MetricType) ([]plugin.MetricType, er
 						log.Warn("Attempted to collect metric ", requested.String(), " but it returned nil!")
 						continue
 					}
+					//clone ns
+					cns := core.Namespace(append([]core.NamespaceElement{}, requested...))
```

Comment:
ok .. here is another explanation if #30 did not help
for object1 = 1..10
  for object2 = 1..2
     put object2 -> object1
     put object1 -> object3  #total = 10 objects 
    #PATCH create objectX = object1+object2, -> object3 = 20 new objects
  end for
end for